healthcare/api-client/v1/datasets/datasets_test.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2020 Google LLC All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import pytest
+import uuid
+
+from googleapiclient import errors
+from retrying import retry
+
+import datasets
+
+cloud_region = 'us-central1'
+project_id = os.environ['GOOGLE_CLOUD_PROJECT']
+
+dataset_id = 'test-dataset-{}'.format(uuid.uuid4())
+destination_dataset_id = 'test-destination-dataset-{}'.format(uuid.uuid4())
+keeplist_tags = 'PatientID'
+time_zone = 'UTC'
+
+
+def retry_if_server_exception(exception):
+    result = False
+    if isinstance(exception, errors.HttpError):
+        if exception.resp.status != 404:
+            logging.warning("Not a 404. Retrying...")
+            result = True
+        else:
+            logging.error("404 response. Not retrying")
+    else:
+        logging.error("404 response. Not retrying")
+    return result
+
+
+@pytest.fixture(scope="module")
+def test_dataset():
+    dataset = datasets.create_dataset(
```

Comment:
Can you add retries for creating/deleting here too?

Reply:
Done.

Reply:
Done. Interestingly I didn't need to yield the dataset to get the fixture value because "dataset" isn't used anywhere.